lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -119,23 +119,23 @@ static StringRef getNameForObjC(const ValueDecl *VD,
   return VD->getName().str();
 }
 
-static Optional<StringRef>
+static bool
 printSwiftEnumElemNameInObjC(const EnumElementDecl *EL, llvm::raw_ostream &OS,
                              Identifier PreferredName = Identifier()) {
   OS << getNameForObjC(EL->getDeclContext()->getAsEnumOrEnumExtensionContext());
   StringRef ElemName = getNameForObjC(EL, CustomNamesOnly);
-  Optional<StringRef> Result = None;
   if (ElemName.empty()) {
     if (PreferredName.empty())
       ElemName = EL->getName().str();
     else
       ElemName = PreferredName.str();
   } else {
-    Result = ElemName;
+    OS << ElemName;
+    return true;
```

Comment:
Much better. :-) Nitpick: Can you turn this into an early exit? And can you add a doc comment to describe the return value?